src/Game/UI/uiFadeProgress.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <heap/seadDisposer.h>
+
+namespace uking::ui {
+
+class FadeProgress {
+    SEAD_SINGLETON_DISPOSER(FadeProgress)
+    FadeProgress() = default;
+    ~FadeProgress() = default;
+
+public:
+    void init();
+    void reset();
+    void updateFadeScreen();
+    void setProgress(float value);
+    float getGaugeValue();
+    void setIsActive();
+
+private:
+    bool mActive = false;
+    f32 mProgress = 0.0;
```

Comment:
Could we use float instead of f32 (for consistency)?

Reply:
As I saw other parts of the code use the defined types in `seadTypes.h`, I thought it was better to use them, but sure.

Reply:
Yeah both float and f32 are used in the codebase as we're pretty bad at being consistent across the entire codebase :sweat_smile:

float is used a lot more though and shows up 5215 times (whereas f32 only shows up 1170 times).

Ultimately I think it doesn't really matter and I think Nintendo probably uses both in their codebase (considering some class names use F32 and others use Float) but it'd be nice to stay consistent within a given source file at least